File: Restaurant-back/src/main/java/com/spring/restaurant/entites/BaseEntity.java
package com.spring.restaurant.entites;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.GenericGenerator;

import java.util.UUID;

@MappedSuperclass
@Getter
@Setter
@NoArgsConstructor
public class BaseEntity
{
	@Id
	@GeneratedValue
	@GenericGenerator(name = "uuid2", strategy = "uuid2")
	@Column(columnDefinition = "BINARY(16)")
	private UUID id;
	private String name;
	private String imagePath;
}

File: Restaurant-front/src/model/BaseEntity.ts
export abstract class BaseEntity {
  id: string;
  name: string;
  imagePath: string;

  constructor(entity?: Partial<BaseEntity>) {
    if (entity) {
      this.id = entity.id;
      this.name = entity.name;
      this.imagePath = entity.imagePath;
    }
  }
}
